### PR TITLE
DEVELOPING.md: specific guidance on profiling

### PR DESCRIPTION
**Pull request by slook**
- Added authoritative reference links to official Python docs, instead of "Search engines".
- Added command syntax help for running `nicotine` with `py-spy`.

### CODE REVIEW

**Comment by mathiascode** [comment]
> - Added authoritative reference links to official Python docs, instead of "Search engines".

Personally, I only use the Python docs for verifying how functions work (and occasionally other useful information there). Most of my time is spent checking things people have come up with in the past on Stack Overflow.

**Comment by slook** [comment]
Well sure me too (though Stack Overflow tends to be a bit above my head, as I only have the attention span for a how-to-geek article or the like :), and I guess that's up to each developer as to how they learn.

I just wanted to drive home the point that we try to do things the "proper" way (i.e. according to the official manual and PEP's etc), rather than just "Googling" everything.

**Comment by mathiascode** [comment]
Sure, but I'd change `help a lot here` to something more neutral in that case. Could you create a separate PR for those changes?

File: doc/DEVELOPING.md [hunk 1]
```diff
@@ -48,9 +48,14 @@ Due to Python's interpreted nature, addressing performance issues can be a chall
 
  * Use different data structures and algorithms, e.g. dictionaries and sets for faster membership checks (O(1)) compared to lists (O(n)).
  * Use existing functionality in the Python Standard Library when available. Parts of the standard library are written in C, and perform better than pure-Python counterparts, especially in hot code paths.
- * Look for alternative ways of accomplishing a task, and measure the performance. Search engines help a lot here.
+ * Look for alternative ways of accomplishing a task, and measure the performance. The [Python documentation](https://docs.python.org) and [Wiki](https://wiki.python.org/moin/PythonSpeed/PerformanceTips) help a lot here.
 
-[py-spy](https://github.com/benfred/py-spy) is an excellent tool for profiling Python applications in real time.
+[py-spy](https://github.com/benfred/py-spy) is an excellent tool for profiling in real time while [running Nicotine+ directly from a local Git folder](https://github.com/nicotine-plus/nicotine-plus/blob/master/doc/TESTING.md#git) and starting with:
```
Comment (mathiascode) [inline doc/DEVELOPING.md:53]:
```suggestion
[py-spy](https://github.com/benfred/py-spy) is an excellent tool for profiling in real time while [running Nicotine+ directly from a local Git folder](TESTING.md#git) and starting with:
```

File: doc/DEVELOPING.md [hunk 2]
```diff
@@ -48,9 +48,14 @@ Due to Python's interpreted nature, addressing performance issues can be a chall
...
+
+```sh
+py-spy top ./nicotine
+```
+The console will continuously display a top like view of functions consuming CPU, press `L` to aggregate by line number.
```
Comment (mathiascode) [inline doc/DEVELOPING.md:58]:
```suggestion
The console will continuously display a top like view of functions consuming CPU. Press `L` to aggregate by line number, and `R` to reset the view.
```

**Pull request merged**